src/libaktualizr/package_manager/packagemanagerconfig.cc

```diff
@@ -9,6 +12,18 @@ void PackageConfig::updateFromPropertyTree(const boost::property_tree::ptree& pt
   CopyFromConfig(ostree_server, "ostree_server", pt);
   CopyFromConfig(packages_file, "packages_file", pt);
   CopyFromConfig(fake_need_reboot, "fake_need_reboot", pt);
+#ifdef BUILD_DOCKERAPP
+  std::string val;
+  CopyFromConfig(val, "docker_apps", pt);
+  if (val.length() > 0) {
+    // token_compress_on allows lists like: "foo,bar", "foo, bar", or "foo bar"
+    boost::split(docker_apps, val, boost::is_any_of(", "), boost::token_compress_on);
+    CopyFromConfig(docker_apps_root, "docker_apps_root", pt);
+    CopyFromConfig(docker_app_params, "docker_app_params", pt);
+    CopyFromConfig(docker_app_bin, "docker_app_bin", pt);
+    CopyFromConfig(docker_compose_bin, "docker_compose_bin", pt);
+  }
+#endif
```

Comment:
Ideally, we wouldn't have package manager-specific configuration options, but I think we need to think a bit more about how to make that a reality, so for now, this is probably fine. We have some examples of config sections that are specific to certain modules, but I don't know if that'd work here because of the package manager design.

Can you provide an example of what the `docker_apps_root` would be in practice? Would you image it being somewhere in `/var`, or should it really be a temporary directory? And you don't seem to use `docker_app_params`; can you explain when it would be helpful? (My apologies if this is obvious to people who use docker-apps more regularly.)

Reply:
> Can you provide an example of what the docker_apps_root would be in practice? Would you image it being somewhere in /var, or should it really be a temporary directory?

It winds up being a directory structure like:
 root/<appX>/{logs.dockerapp,docker-compose.yml}

I really just wanted this to use something relative to StorageConfig::path and the configuration I set for everyone has been /var/sota/docker-apps. I didn't see a great way to infer this that was very clean. I'm open to suggestions. Maybe we could do something like this:

1) Change this directory to always be compiled in and be named `PackageManagerConfig::packages_dir`.
2) Change `Config::postUpdateValues` to always set this relative to StorageConfig::path?

Reply:
> I really just wanted this to use something relative to StorageConfig::path and the configuration I set for everyone has been /var/sota/docker-apps. I didn't see a great way to infer this that was very clean. I'm open to suggestions.

Now I understand. We have the same issue with `import.base_path`. Currently it is actually independent of `storage.path` but in practice we never change the defaults, which are `/var/sota/import` and `/var/sota`, respectively. It probably makes sense to copy that style for now, but I'm open to discussion on that point. Either of your alternate suggestions could work, but I want to keep custom logic in the Config class to a minimum if possible.